packages/core/src/browser/fetchObservable.ts

```diff
@@ -113,5 +116,61 @@ function afterSend(
       observable.notify(context)
     }
   }
-  responsePromise.then(monitor(reportFetch), monitor(reportFetch))
+
+  const reportFetchResolve = (response: Response) => {
+    let stopInstruments: Array<() => void> = []
+    let timeOutId: null | number = null
+
+    Promise.race([
+      new Promise((resolve) => {
+        stopInstruments = (['json', 'text', 'formData', 'blob', 'arrayBuffer'] as OverloadMethodKeys[]).map(
+          (method) => {
+            const { stop } = instrumentMethod(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Maybe a new helper function `instrumentMethods(object: OBJECT,
  methods: METHOD[],...` could help simplifying

Reply:
great idea!